components/my-account/User/UserForm/UserForm.tsx

```diff
@@ -0,0 +1,368 @@
+                  }}
+                >
+                  <ClearIcon />
+                </Button>
+                <Button
+                  variant="contained"
+                  disableElevation
+                  style={{ height: '37px' }}
+                  type="submit"
+                >
+                  {isLoading ? (
+                    <CircularProgress
+                      style={{
+                        color: theme.palette.secondary.light,
+                        height: '25px',
+                        width: '25px',
+                      }}
+                    />
+                  ) : (
+                    <CheckIcon />
+                  )}
+                </Button>
+              </>
+            )}
+            {!isEditMode && mdScreen && (
+              <>
+                <Button
+                  variant="outlined"
+                  type="reset"
+                  onClick={closeUserForm}
+                  style={{ marginRight: '8px' }}
+                >
+                  {t('cancel')}
+                </Button>
+                <Button
+                  variant="contained"
+                  disableElevation
+                  style={{ height: '37px', width: '90px' }}
+                  type="submit"
+                >
+                  {isLoading ? (
+                    <CircularProgress
+                      size={'small'}
+                      style={{
+                        color: theme.palette.secondary.light,
+                        height: '25px',
+                        width: '25px',
+                      }}
+                    />
+                  ) : (
+                    t('add-user')
+                  )}
+                </Button>
+              </>
+            )}
+
+            {!mdScreen && (
+              <Grid
+                style={{
+                  display: 'flex',
+                  justifyContent: 'center',
+                  flexDirection: 'column',
+                  width: '100%',
+                }}
+              >
+                <Button
+                  variant="contained"
+                  disableElevation
+                  style={{ height: '37px', marginBottom: '10px' }}
+                  type="submit"
+                >
+                  {isLoading ? (
+                    <CircularProgress
+                      size={'small'}
+                      style={{
+                        color: theme.palette.secondary.light,
+                        height: '25px',
+                        width: '25px',
+                      }}
+                    />
+                  ) : (
+                    t('save')
+                  )}
+                </Button>
+                <Button variant="outlined" type="reset" onClick={closeUserForm}>
+                  {t('cancel')}
+                </Button>
+              </Grid>
+            )}
+          </Grid>
+        </Grid>
+      </form>
+    </>
+  )
+}
+
+export default UserForm
```

Comment:
**1. Story is missing**

**2. Testcase are missing**

**3. change closeUserForm prop to onClose**
const { isEditMode, b2BUser, closeUserForm } = props

**4. missing prop onSave**
save/edit functionality should handle by parent.

**5. Can we destructure instead of calling twice?**
  const userRoles = publicRuntimeConfig.b2bUserRoles
  const userFormRadioOptions = publicRuntimeConfig.userFormRadioOptions

6. Avoid useEffect for state syncing
No need of useEffect here

```
useEffect(() => {
    if (!b2BUser) return
    const { firstName, lastName, emailAddress, isActive, roles } = b2BUser
    setValue('emailAddress', emailAddress)
    setValue('firstName', firstName)
    setValue('lastName', lastName)
    setValue('isActive', isActive)
    setValue('role', roles && roles.length ? roles[0]?.roleName : '')
  }, [b2BUser])
```

Reply:
For point 5
it was throwing error when tried to destructure